m365-developer-proxy-plugins/Inspection/DevToolsPlugin.cs

```diff
@@ -0,0 +1,299 @@
+        {
+            Body = string.Empty,
+            Base64Encoded = false
+        };
+        if (IsTextResponse(e.Session.HttpClient.Response.ContentType))
+        {
+            body.Body = e.Session.HttpClient.Response.BodyString;
+            body.Base64Encoded = false;
+        }
+        else
+        {
+            body.Body = Convert.ToBase64String(e.Session.HttpClient.Response.Body);
+            body.Base64Encoded = true;
+        }
+        responseBody.Add(e.Session.HttpClient.Request.GetHashCode().ToString(), body);
+
+        var requestId = GetRequestId(e.Session.HttpClient.Request);
+
+        var responseReceivedMessage = new ResponseReceivedMessage
+        {
+            Params = new()
+            {
+                RequestId = requestId,
+                LoaderId = "1",
+                Timestamp = (double)DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() / 1000,
+                Type = "XHR",
+                Response = new()
+                {
+                    Url = e.Session.HttpClient.Request.Url,
+                    Status = e.Session.HttpClient.Response.StatusCode,
+                    StatusText = e.Session.HttpClient.Response.StatusDescription,
+                    Headers = e.Session.HttpClient.Response.Headers.Select(h =>
+                        new KeyValuePair<string, string>(h.Name, h.Value)).ToDictionary(kvp => kvp.Key, kvp => kvp.Value),
+                    MimeType = e.Session.HttpClient.Response.ContentType
+                }
+            }
+        };
+
+        await webSocket.SendAsync(responseReceivedMessage);
+
+        var loadingFinishedMessage = new LoadingFinishedMessage
+        {
+            Params = new()
+            {
+                RequestId = requestId,
+                Timestamp = (double)DateTimeOffset.UtcNow.ToUnixTimeMilliseconds() / 1000,
+                EncodedDataLength = e.Session.HttpClient.Response.HasBody ? e.Session.HttpClient.Response.Body.Length : 0
+            }
+        };
+        await webSocket.SendAsync(loadingFinishedMessage);
+    }
+
+    private bool IsTextResponse(string? contentType)
+    {
+        var isTextResponse = false;
+
+        if (contentType is not null &&
+            (contentType.IndexOf("text") > -1 ||
+            contentType.IndexOf("json") > -1))
+        {
+            isTextResponse = true;
+        }
+
+        return isTextResponse;
+    }
+
+    private async void AfterRequestLog(object? sender, RequestLogArgs e)
+    {
+        if (webSocket?.IsConnected != true ||
+            e.RequestLog.MessageType == MessageType.InterceptedRequest ||
+            e.RequestLog.MessageType == MessageType.InterceptedResponse)
+        {
+            return;
+        }
+
+        var message = new EntryAddedMessage
+        {
+            Params = new()
+            {
+                Entry = new()
+                {
+                    Source = "network",
+                    Text = string.Join(" ", e.RequestLog.Message),
```

Comment:
*nit:* perhaps we should rename this `Message` property to be `Messages` or `MessageLines`?

Reply:
MessageLines would be clearer. Agreed